### PR TITLE
chore: update to angular rc3 & newest router

### PR DESCRIPTION
**Pull request by jelbourn**
R: @hansl or @robertmesserle 

Need the new router to work with offline compiler.

### CODE REVIEW

**Comment by jelbourn** [comment]
CI failure is due to https://github.com/angular/vladivostok/issues/95

File: src/demo-app/index.html [hunk 1]
```diff
@@ -5,7 +5,7 @@
   <meta charset="utf-8">
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <title>Material2</title>
-  <base href="/">
+  <base href=".">
```
Comment (robertmesserle) [inline src/demo-app/index.html]: Why is this necessary?

**Comment by hansl** [comment]
LGTM except for @robertmesserle's comment.

**Comment by hansl** [comment]
And we probably want to wait for that router issue to be fixed since master will become red.

**Comment by jelbourn** [comment]
Yeah, waiting on Victor to fix the circular dep in the router.

File: src/demo-app/index.html (hunk 1 above)
Comment (jelbourn) [inline src/demo-app/index.html]: Angular requires an explicit base tag, and [the docs recommend this value](https://angular.io/docs/ts/latest/guide/router.html).

**Comment by zigzag95** [comment]
Please be careful because base href="." May break apps using webpack.
Why are you using v3 router?

**Comment by hansl** [comment]
We have constant testing with webpack currently. Router v3 is ready for simple cases and Material2 is a good test bed for it.

**Comment by jelbourn** [comment]
@zigzag95 This use of the router is only for our own demo app and doesn't affect any consumers of the components.

**Comment by zigzag95** [comment]
So It must be a peer depency or dev depency

**Comment by ddomingues** [comment]
Hi @zigzag95 and @jelbourn
 
Wouldn't be interesting already update it to angular rc3?

**Comment by zigzag95** [comment]
I think WIP.
